botfront/imports/ui/components/nlu/models/NLUModel.jsx

```diff
@@ -165,35 +164,44 @@ class NLUModel extends React.Component {
         if (instance) {
             tabs.push({
                 menuItem: 'API',
-                render: () => (
-                    <React.Suspense fallback={<Loader active />}>
-                        <API model={model} instance={instance} />
-                    </React.Suspense>
-                ),
+                render: () => (<API model={model} instance={instance} />),
             });
         }
         return tabs;
     };
 
     getSettingsSecondaryPanes = () => {
-        const { model, instances, intents } = this.props;
+        const { model, intents, projectDefaultLanguage } = this.props;
+        const cannotDelete = model.language !== projectDefaultLanguage;
         return [
-            { menuItem: 'General', render: () => <NLUParams model={model} instances={instances} /> },
             { menuItem: 'Pipeline', render: () => <NLUPipeline model={model} onSave={this.onUpdateModel} /> },
             { menuItem: 'Import', render: () => <DataImport model={model} /> },
             { menuItem: 'Export', render: () => <DataExport intents={intents} model={model} /> },
-            { menuItem: 'Delete', render: () => <DeleteModel model={model} onDeleteModel={this.onDeleteModel} /> },
+            { menuItem: 'Delete', render: () => <DeleteModel model={model} onDeleteModel={this.onDeleteModel} cannotDelete={cannotDelete} /> },
         ];
     };
 
+    onChange = (e, data) => {
```

Comment:
don't just call it onChange, we don't know what's changing.
Methods that update controlled components should be called handle<something>
Here you could call the method handleLanguageChange() for instance

Reply:
Done, I forgot about the convention. Changed it to handleLanguageChange